### PR TITLE
fix: links in article content

### LINKED ISSUE #2901: Issue with all content: Missing link underlines
MDN URL: https://developer.mozilla.org/en-US/docs/ (and all pages below)

#### Issue

None of the links within the content have an underline. They are difficult to distinguish from the surrounding content and require interaction to identify them in some cases.

The page content is dark gray (`#212121`) and the link color is dark blue (`#00458b`). with a contrast ratio of 1.7:1, it is fair to assume most users with good vision under ideal conditions will have to take a second look.

As it stands now, the content links would fail WCAG 2.1 [Success Criterion 1.4.1 Use of Color](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html). The non-normative [G182](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20160317/G182) recommends using an additional visual cue besides color. [G183](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20160317/G183.html) recommends a 3:1 contrast ratio and an additional visual cue on focus or hover, but is now mooted with the advent of touch interfaces. I go into more detail about alternatives to underlines in [_On Link Underlines_](https://adrianroselli.com/2016/06/on-link-underlines.html).

#### Recommended fix

To restore underlines to links in the content…

In the file:
/node_modules/@mdn/minimalist/sass/atoms/_typography.scss

At line 69, please strike `text-decoration: none;`. Reference code with - preceding the line to strike:

```diff
  /* text links */
  a:link,
  a:visited {
    color: $link-color;
-    text-decoration: none;

    &:hover,
    &:focus,
    &:active {
      text-decoration: underline;
    }
  }
```

In the file:
/node_modules/@mdn/minimalist/sass/atoms/_links.scss

At line 3, please strike `text-decoration: none;`. Reference code with - preceding the line to strike:

```diff
  a {
    color: $link-color;
-    text-decoration: none;

    &:hover,
    &:focus {
      text-decoration: underline;
    }
```

This will restore underlines to links in the navigation and elsewhere as well. If you want to remove those, I suggest more targeted removal of underlines scoped to just those areas to ensure underlines are retained on links outside the content.

If you would like to try the content with restored underlines without fiddling in the browser's dev tools, I have bookmarklet that will restore underlines (which I tested on MDN) at http://rosel.li/bookmarklets#underlines.

#### Specific section or headline?

Throughout.

#### What did you expect to see?

Underlines on links.

#### Did you test this? If so, how?

I tried to use the content.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/element`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/web/html/element/index.html
* Last commit: https://github.com/mdn/content/commit/145e9f6f3d388fd2277aaf47f0b68175e4f9dcb5
* Document last modified: 2020-12-18T22:31:45.000Z

</details>

### PR DESCRIPTION
**Pull request by schalkneethling**
Fixes underlines and colors of links in main page content.

Fix #2901

### CODE REVIEW

**Review by peterbe — COMMENTED** [review]
Perhaps there's a good reason for it but I find it so odd that the sidebars, breadcrumbs, document-footer are not underlined, but all the links within the document itself are. 
I think it makes sense for the links in the top navbar to not conform because they're almost expected to be different. 

And why aren't the links on the home page underlined? Or the writer's home page? Or the 404 page?

I don't mind the inconsistencies if I understood why it is like that. 

Mind you, I don't know much about CSS resets but why can't it just be that all links are underlined, *except* the exceptions like the navbar, auto-complete search results, buttons, and maybe the page footer (because they're of an entirely different color). 

Also, I admit that I don't understand a11y very well so perhaps there's a story there. But right now it feels a bit "hotch-potch mixed bag".

File: client/src/document/index.scss [hunk 1]
```diff
@@ -36,7 +36,7 @@
   }
 }
 
-.article {
+.main-page-content {
```
Comment (peterbe) [inline client/src/document/index.scss:39]: You're going to need to fix the print stylesheet in public/index.html too now.

**Comment by schalkneethling** [comment]
> Perhaps there's a good reason for it but I find it so odd that the sidebars, breadcrumbs, document-footer are not underlined, but all the links within the document itself are.

So, here is my thinking. It is a delicate balance one has to achieve here. In some contexts such as in the sidebar, the header, and the footer I believe people pretty much assume the items in there to be links in general.

The reason the underline is important in the main article body is that while the color contrast between the link text and the background is good, the contrast between the plain text and the link text is not as clear. Adding explicit underlining to these then makes them clearly stand out.

If we add link underlines to everything wholesale, you will quickly get into a situation where we cause a different problem in terms of cognitive overload. There is just too much asking for your attention. Our main content already suffers from this in that the articles feel a bit like a wall of text.

This is something that we will address next.

**Comment by peterbe** [comment]
> > Perhaps there's a good reason for it but I find it so odd that the sidebars, breadcrumbs, document-footer are not underlined, but all the links within the document itself are.
> 
> So, here is my thinking. It is a delicate balance one has to achieve here. In some contexts such as in the sidebar, the header, and the footer I believe people pretty much assume the items in there to be links in general.
> 
> The reason the underline is important in the main article body is that while the color contrast between the link text and the background is good, the contrast between the plain text and the link text is not as clear. Adding explicit underlining to these then makes them clearly stand out.
> 
> If we add link underlines to everything wholesale, you will quickly get into a situation where we cause a different problem in terms of cognitive overload. There is just too much asking for your attention. Our main content already suffers from this in that the articles feel a bit like a wall of text.
> 
> This is something that we will address next.

Ok. That's a great argument for not using underlines in the sidebar, because, apart from headings, everything is a link. Same is true for most things in the footer and the top navbar. But what about the home page, the document footer, apps. They're also blue texts on white background next to lots of non-link black texts on white background. 

I'm wondering, would it be a better idea to make underlining the *default* and then sidebars and top navbar become the *exception*.

**Comment by schalkneethling** [comment]
> But what about the home page

I do not think there is anything on the homepage that would really benefit from being underlined. The links that are there are clearly distinguishable 

> I'm wondering, would it be a better idea to make underlining the default and then sidebars and top navbar become the exception.

I reckon we will be overriding a lot more than it might immediately seem. I am happy to be wrong, but I reckon we are in a good place with this PR.

> document footer

I assume you are referring to "Found a problem with this page?"? I also thought about it and I reckon it is a good idea to underline those links as well. Let me update the PR with that.

**Comment by schalkneethling** [comment]
@peterbe updated to also underline links in the document footer. Thank you for the suggestion.

**Review by peterbe — APPROVED** [review]

**Pull request merged**